WordPress/Classes/ViewRelated/Tools/PromptViewController.swift

```diff
@@ -0,0 +1,123 @@
+import Foundation
+
+
+/// Protocol that defines the methods required by PromptViewController's Children(s).
+///
+@objc
+protocol PresentedViewController
+{
+    var doneButtonEnabled : Bool { get }
+    func cancelButtonWasPressed(sender: AnyObject)
+    func doneButtonWasPressed(sender: AnyObject)
+}
+
+
+/// ViewController container, that presents a Done / Cancel button, and forwards their events to
+/// the childrenViewController (which *must* implement PresentedViewController).
+///
+class PromptViewController : UIViewController
+{
+    deinit {
+        stopListeningToProperties(childrenViewController)
+    }
+    
+    init(viewController: UIViewController) {
+        super.init(nibName: nil, bundle: nil)
+        assert(viewController.conformsToProtocol(PresentedViewController))
+        
+        setupNavigationButtons()
+        attachChildrenViewController(viewController)
+        setupChildrenViewConstraints(viewController.view)
+        startListeningToProperties(viewController)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
```

Comment:
Wow. The compiler is *super* smart. It complaints, *unless* there's a **fatalError** statement.

**Wow*!!!